pds/contracts/PDS.cdc

```diff
@@ -112,52 +149,56 @@ pub contract PDS{
                 i = i + 1
             }
             c.open(id: packId, nfts: nfts)
-            PDS.releaseEscrow(nftIds: toReleaseNFTs, recvCap: recvCap , collectionProviderPath: collectionProviderPath)
+            PDS.releaseEscrow(nftIds: toReleaseNFTs, recvCap: recvCap , collectionStoragePath: collectionStoragePath)
         }
 
-
-        init(
-            withdrawCap: Capability<&{NonFungibleToken.Provider}>,
-            operatorCap: Capability<&{IPackNFT.IOperator}>
-
-        ){
+        /// SharedCapabilities resource initializer.
+        ///
+        view init(
+            withdrawCap: Capability<auth(NonFungibleToken.Withdraw, NonFungibleToken.Owner) &{NonFungibleToken.Provider}>,
+            operatorCap: Capability<auth(IPackNFT.Operatable) &{IPackNFT.IOperator}>
+        ) {
             self.withdrawCap = withdrawCap
             self.operatorCap = operatorCap
         }
     }
 
 
-    pub resource interface PackIssuerCapReciever {
-        pub fun setDistCap(cap: Capability<&DistributionCreator{IDistCreator}>)
-    }
+    // Included for backwards compatibility.
+    access(all) resource interface PackIssuerCapReciever {}
 
-    pub resource PackIssuer: PackIssuerCapReciever {
-        access(self) var cap:  Capability<&DistributionCreator{IDistCreator}>?
+    /// Resource that defines the issuer of a pack.
+    ///
+    access(all) resource PackIssuer: PackIssuerCapReciever {
+        access(self) var cap: Capability<&DistributionCreator>?
 
-        pub fun setDistCap(cap: Capability<&DistributionCreator{IDistCreator}>) {
+        access(all) fun setDistCap(cap: Capability<&DistributionCreator>) {
```

Comment:
Should this function check that self.cap is `nil` in a pre-condition?

Reply:
Good call! Fixed this 👍